debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsStreamProducer.java

```diff
@@ -127,6 +130,9 @@ protected synchronized void start(BlockingConsumer<ChangeEvent> eventConsumer, C
 
     private void streamChanges(BlockingConsumer<ChangeEvent> consumer, Consumer<Throwable> failureConsumer) {
         ReplicationStream stream = this.replicationStream.get();
+        // once we are streaming changes, we stop the keep alive, as the read loop
+        // will ensure that happens
+        stream.stopKeepAlive();
```

Comment:
Will this be invoked if the connector stops early (i.e. is shut down before the snapshot is complete)? We must make sure no threads are leaked in such case.

Reply:
@addisonj, I still think there's a (slight) risk of leaking that keep-alive thread under unfortunate circumstances. It seems that `streamChanges()` may be interrupted (by calling `stop()`), before the call flow gets here. I think to be safe `stop()` itself should call `stream.stopKeepAlive()` (which would have to be idempotent for that). WDYT?

@jpechane, how do you feel about this change, does it interfere much with your ongoing work on the connector unification?